### DISCUSSION TITLE
VsCode Deno Can't Find Supabase Module

### DISCUSSION THREAD

**Discussion by onurhan1337** [supabase/supabase#27149] [Questions]
Hey, very new to Supabase, and Deno. I'm testing out edge function running locally on machine. Has anyone gotten type checking to work for the Supabase module? I'm trying to set up a session, and it works if I run it, but can't get type checking and intellisense to sense the Supabase module.

```ts
// Setup type definitions for built-in Supabase Runtime APIs
/// <reference types="https://esm.sh/@supabase/functions-js/src/edge-runtime.d.ts" />

import { createClient } from "@supabase/supabase-js";
import { Database } from "../_lib/database.ts";

const model = new Supabase.ai.Session("gte-small");
```

![image](https://github.com/supabase/supabase/assets/60575283/a1355464-612b-4ebd-8444-cf0080e22817)

I'm experimenting with them based on the Supabase Community project. I had no problem with the first 2 phases, but in phase 3 embeeding I get this error and so I can't move forward.

https://github.com/supabase-community/chatgpt-your-files/

Also, if anyone has already browsed the repo I shared above, I made it to Chapter 4 but I encountered this problem in 3. The process file in supabase/functions was actually performing its function correctly, but then when I developed the embed file, I started getting the error in the picture in docker. The file loads without any problems, but it gives an error where it will be split into sections.

![image](https://github.com/supabase/supabase/assets/60575283/42f79492-9ff2-4b58-95d5-c9f8037b5db0)

**Comment by adgent** [comment]
Encountering the same error

**Comment by cpannwitz** [comment]
This problem still persists for us.
While being in VSCode, the types are found correctly. But using `deno lint && deno check`, the types are not found.

```
/// <reference types="jsr:@supabase/functions-js/edge-runtime.d.ts" />

import "jsr:@supabase/functions-js/edge-runtime.d.ts";

export function createSupabaseRuntimeAISession() {
  return new Supabase.ai.Session("gte-small");
}
```
